packages/jest-runtime/src/index.ts

```diff
@@ -526,12 +521,18 @@ class Runtime {
     };
   }
 
-  requireModuleOrMock(from: Config.Path, moduleName: string): unknown {
+  requireModuleOrMock<T = unknown>(from: Config.Path, moduleName: string): T {
+    // this module is unmockable
+    if (moduleName === '@jest/globals') {
```

Comment:
noticed `automock` in the test broke this, so moved the import to before checking for mocks

Reply:
I've made sure this fix is part of revert of the revert, btw